packages/@aws-cdk/aws-codepipeline-actions/lib/cloudformation/pipeline-actions.ts

```diff
@@ -513,6 +513,487 @@ export class CloudFormationDeleteStackAction extends CloudFormationDeployAction
+
+  /**
+   * Stack resources are retained when an account is removed from a target organization or OU
+   */
+  ENABLED_WITH_STACK_RETENTION = 'EnabledWithStackRetention'
+}
+
+/**
+ * A StackSet parameter
+ */
+export interface StackSetParameter {
+  /**
+   * The name of the parameter
+   */
+  readonly parameterKey: string
+
+  /**
+   * The parameter value.
+   *
+   * @default No default value
+   */
+  readonly parameterValue?: string
+
+  /**
+   * Use previous value? If this is specified, parameterValue needn't be specified for a StackSet update
+   *
+   * @default false
+   */
+  readonly usePreviousValue?: boolean
+}
+
+/**
+ * Properties for the CloudFormationStackSetAction
+ */
+export interface CloudFormationStackSetActionProps extends codepipeline.CommonAwsActionProps {
+  /**
+   * The AWS region the given Action resides in.
+   * Note that a cross-region Pipeline requires replication buckets to function correctly.
+   * You can provide their names with the {@link PipelineProps#crossRegionReplicationBuckets} property.
+   * If you don't, the CodePipeline Construct will create new Stacks in your CDK app containing those buckets,
+   * that you will need to `cdk deploy` before deploying the main, Pipeline-containing Stack.
+   *
+   * @default the Action resides in the same region as the Pipeline
+   */
+  readonly region?: string;
+
+  /**
+   * The name to associate with the stack set. This name must be unique in the Region where it is created.
+   *
+   * The name may only contain alphanumeric and hyphen characters. It must begin with an alphabetic character and be 128 characters or fewer.
+   */
+  readonly stackSetName: string;
+
+  /**
+   * A description of the stack set. You can use this to describe the stack set’s purpose or other relevant information.
+   *
+   * @default No description
+   */
+  readonly description?: string;
+
+  /**
+   * The location of the template that defines the resources in the stack set. This must point to a template with a maximum size of 460,800 bytes.
+   *
+   * Enter the path to the source artifact name and template file.
+   */
+  readonly templatePath: codepipeline.ArtifactPath;
+
+  /**
+   * A list of template parameters for your stack set that update during a deployment.
+   * You can provide parameters as a dictionary or a file path:
+   *
+   * The following example shows a parameter named BucketName with the value my-bucket.
+   *
+   * [ { ParameterKey: "BucketName", ParameterValue: "my-bucket" } ]
+   *
+   * The following example shows an entry with multiple parameters:
+   *
+   * [
+   *     {
+   *         ParameterKey: "BucketName",
+   *         ParameterValue: "my-bucket"
+   *     },
+   *     {
+   *         ParameterKey: "Asset1",
+   *         ParameterValue: "true"
+   *     },
+   *     {
+   *         ParameterKey: "Asset2",
+   *         ParameterValue: "true"
+   *     }
+   * ]
+   *
+   * This cannot be used in conjunction with parametersPath.
```

Comment:
I don't like these mutually-exclusive properties. Let's instead merge these into one:

```ts
export interface CloudFormationStackSetActionProps extends codepipeline.CommonAwsActionProps {
  readonly parameters?: StackSetParameters;

  // ...
}

export class StackSetParameters {
  public static fromLiteral(parameters: { [parameterName: string]: string? }): StackSetParameters {
    // ...
  }

  public static fromArtifactPath(artifactPath: codepipelne.ArtifactPath): StackSetParameters {
    // ...
  }

  private constructor() {
    // ...
  }
}
```

We call this the **union-like pattern** in the CDK.